File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/tag/AtomTag.kt
package com.perrystreet.woof.designsystem.atomic.atoms.tag

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic.atoms.tag.roles.TagStyleRole
import com.perrystreet.woof.designsystem.theme.Theme

@Composable
fun AtomTag(
    text: String,
    styleRole: TagStyleRole = TagStyleRole.Neutral,
) {
    Text(
        text = text,
        style = Theme.typography.subhead.p3,
        color = styleRole.textColor(),
        maxLines = 1,
        modifier = Modifier
            .background(
                color = styleRole.backgroundColor(),
                shape = RoundedCornerShape(Theme.sizing.radiusXL),
            )
            .padding(
                horizontal = Theme.padding.elementRelaxed,
                vertical = Theme.padding.elementCompact,
            ),
    )
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Tag/AtomTag.swift
import SwiftUI

public struct AtomTag: View {
    @Environment(\.theme) private var theme

    private let text: String
    private let styleRole: TagStyleRole

    public init(text: String, styleRole: TagStyleRole = .neutral) {
        self.text = text
        self.styleRole = styleRole
    }

    public var body: some View {
        Text(text)
            .font(theme.typography.subhead.p3)
            .foregroundStyle(styleRole.textColor(from: theme))
            .lineLimit(1)
            .padding(.horizontal, PaddingRoles.Element.relaxed.rawValue)
            .padding(.vertical, PaddingRoles.Element.compact.rawValue)
            .background(styleRole.backgroundColor(from: theme), in: RoundedRectangle(cornerRadius: theme.radius.xl))
    }
}
